core/unit_test/TestViewCtorDimMatch.hpp

```diff
@@ -47,384 +47,150 @@
 
 namespace Test {
 
-#define LIVE(EXPR, ARGS, DYNRANK) EXPECT_NO_THROW(EXPR)
-#define DIE(EXPR, ARGS, DYNRANK)                                           \
-  ASSERT_DEATH(                                                            \
-      EXPR,                                                                \
-      "Constructor for Kokkos View 'v_" #ARGS                              \
-      "' has mismatched number of arguments. Number of arguments = " #ARGS \
-      " but dynamic rank = " #DYNRANK)
+#define LIVE(EXPR, ARGS, DYNRANK, TOTALRANK) EXPECT_NO_THROW(EXPR)
+#define DIE(EXPR, ARGS, DYNRANK, TOTALRANK)                                   \
+  ASSERT_DEATH(                                                               \
+      EXPR,                                                                   \
+      "Constructor for Kokkos::View 'v' has mismatched number of arguments. " \
+      "The number of arguments = " +                                          \
+          std::to_string(ARGS) +                                              \
+          " neither matches the dynamic rank = " + std::to_string(DYNRANK) +  \
+          " nor the total rank = " + std::to_string(TOTALRANK))
+
+template <int rank, int dynrank, template <int> class RankType,
+          std::size_t... args>
+void test_matching_arguments_rank_helper(std::index_sequence<args...>) {
+  constexpr int nargs = sizeof...(args);
+  using view_type     = Kokkos::View<typename RankType<rank>::type>;
+  if (nargs == rank || nargs == dynrank)
+    LIVE({ view_type v("v", args...); }, nargs, dynrank, rank);
+  else
+    DIE({ view_type v("v", args...); }, nargs, dynrank, rank);
+}
 
-#define PARAM_0
-#define PARAM_1 1
-#define PARAM_2 1, 1
-#define PARAM_3 1, 1, 1
-#define PARAM_4 1, 1, 1, 1
-#define PARAM_5 1, 1, 1, 1, 1
-#define PARAM_6 1, 1, 1, 1, 1, 1
-#define PARAM_7 1, 1, 1, 1, 1, 1, 1
+template <int rank, int dynrank, template <int> class RankType>
+void test_matching_arguments_rank() {
+  test_matching_arguments_rank_helper<rank, dynrank, RankType>(
+      std::index_sequence<>());
```

Comment:
```suggestion
      std::make_index_sequence<N>());
```

Reply:
You'd need to use a fold expression to make them all 1
(Not critical change if you dont agree)